SparkPost/2web2ui
# FE-988 - Hibana banner and theme toggling UX enhancements

[FE-988](https://jira.int.messagesystems.com/browse/FE-988)

### What Changed
- Updates `HibanaContext` to handle disabling the Hibana banner, enabling/disabling the Hibana theme, and the `hasThemeControls` state.
- Updates `HibanaBanner` to meet new UX requirements (see the mocks in the ticket)
- Added an `index.js` for the `ToggleBlock` component - updated imports of `ToggleBlock` elsewhere
- Refactors `ProfilePage` component from a class to a function
- Note that `useHibana` still returns an array unnecessarily - this _could_ be cleaned up - but let's talk about it. All of this code is temporary at this point, and this would touch a _lot_ of files. Fairly easy to make the change, but pretty tough for a review.

### How To Test
1. Sign up for a new account, and turn update the **account** UI options with:
```
{ "hibana": { "hasThemeControls": true } }
```
2. Go to the profile page and verify the theme controls render
3. Return to the dashboard page
4. Click on "Take a look!" and verify you are navigated to the profile page, the banner goes away, and the **user's** UI options are updated with `{ "isHibanaBannerVisible": false }`
5. Click on the theme toggle and verify the theme updates. Sign in to your account using a different browser and verify the theme selection has persisted.

1. Make a new account and update the **account** UI options as you did in the previous steps
2. Click the dismiss X inside the banner and verify relevant UI options are updated and the banner goes away

- Make sure integration tests capture relevant scenarios!
- Verify that Pendo is receiving accurate events when the user makes a theme change

### To Do
- [ ] Update feature flags doc:  https://confluence.int.messagesystems.com/display/ENG/FE%3A+Feature+Flags
